### PR TITLE
[REVIEW] Fix for column name issue coming out of Groupby objects

### LINKED ISSUE #934: [BUG] Groupby objects aren't indexable by column names
```python
import cudf
df = cudf.DataFrame({'xx': [1., 2., 3.], 'yy': [1, 2, 3]})
g = df.groupby('yy')
g['xx']
```

```python-traceback
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-4-79b3988cb6ff> in <module>()
----> 1 g['xx']

~/cudf/python/cudf/groupby/groupby.py in __getitem__(self, arg)
    264         for val in arg:
    265             if val not in self._val_columns:
--> 266                 raise KeyError("Column not found: " + val)
    267         self._val_columns = arg
    268         return self

KeyError: 'Column not found: x'
```

This is because we always iterate over the indexer, even if it is a string

```python
    def __getitem__(self, arg):
        for val in arg:
            if val not in self._val_columns:
                raise KeyError("Column not found: " + val)
        self._val_columns = arg
        return self
```

A small type check there would probably resolve the problem.

cc @thomcom

### PR DESCRIPTION
**Pull request by thomcom**
This fixes #934.

Handle strings better throughout Groupby new logic.

### CODE REVIEW

File: python/cudf/groupby/groupby.py [hunk 1]
```diff
@@ -139,6 +139,8 @@ def _apply_agg(self, agg_type, result, add_col_values,
         need_to_index = self._as_index
 
         col_count = 0
+        if isinstance(val_columns, str):
```
Comment (kkraus14) [inline python/cudf/groupby/groupby.py]: What if the column is an integer?

File: python/cudf/groupby/groupby.py [hunk 2]
```diff
@@ -204,8 +206,11 @@ def _apply_agg(self, agg_type, result, add_col_values,
             out_col_agg_series.data.size = num_row_results
             out_col_agg_series = out_col_agg_series.reset_index()
 
-            result[val_columns_out[col_count]
-                   ] = out_col_agg_series[:num_row_results]
+            if isinstance(val_columns_out, str):
```
Comment (kkraus14) [inline python/cudf/groupby/groupby.py]: Same here.

File: python/cudf/groupby/groupby.py (hunk 1 above)
Comment (thomcom) [reply python/cudf/groupby/groupby.py]: That sounds really important!

File: python/cudf/groupby/groupby.py [hunk 3]
```diff
@@ -239,7 +244,7 @@ def _apply_basic_agg(self, agg_type, sort_results=False):
 
         # If a Groupby has one index column and one value column
         # and as_index is set, return a Series instead of a df
-        if len(val_columns) == 1 and self._as_index:
+        if isinstance(val_columns, str) and self._as_index:
```
Comment (kkraus14) [inline python/cudf/groupby/groupby.py]: here too

File: python/cudf/groupby/groupby.py [hunk 4]
```diff
@@ -253,17 +258,25 @@ def _apply_basic_agg(self, agg_type, sort_results=False):
...
 
         nvtx_range_pop()
 
         return result
 
     def __getitem__(self, arg):
-        for val in arg:
-            if val not in self._val_columns:
-                raise KeyError("Column not found: " + val)
+        if isinstance(arg, str):
```
Comment (kkraus14) [inline python/cudf/groupby/groupby.py]: here too

**Comment by kkraus14** [comment]
rerun tests

File: python/cudf/groupby/groupby.py (hunk 1 above)
Comment (thomcom) [reply python/cudf/groupby/groupby.py]: Hmm I'm not sure we ever supported integer column names for groupby. Looking into it.

**Comment by thomcom** [comment]
Separating out sequence vs. string vs. Number is a significant refactor.

File: python/cudf/groupby/groupby.py (hunk 1 above)
Comment (thomcom) [reply python/cudf/groupby/groupby.py]: Fixed, ready for merge.

**Pull request merged**